skyportal/handlers/api/internal/annotations_info.py

```diff
@@ -32,29 +50,46 @@ def get(self):
         # filters to apply on the auto-annotations column on the scanning page.
         # For example, if given that an annotation field is numeric we should
         # have min/max fields on the form.
-        annotations = func.jsonb_each(Annotation.data).table_valued("key", "value")
-
-        with self.Session() as session:
-            # Objs are read-public, so no need to check that annotations belong to an unreadable obj
-            # Instead, just check for annotation group membership
-            results = session.execute(
-                Annotation.select(session.user_or_token, columns=[Annotation.origin])
-                .add_columns(
-                    annotations.c.key,
-                    func.jsonb_typeof(annotations.c.value).label("type"),
+
+        try:
```

Comment:
Given the number of times we are doing this construction, how hard would it be to turn this into a function? not necessarily for this PR, but maybe worth thinking about.

Reply:
To load things? Yeah I thought about the exact same thing while writing this 😅. Given that it will be helpful to get this in before deploy, I suggest we merge it as is, and I'll be happy to open a separate PR this week to add load methods to the cache utils.